packages/api/src/server/permission/service/permission.factory.ts

```diff
@@ -45,10 +45,12 @@ export class PermissionFactory {
     const emails = Array.from(new Set(permissions.map((item) => item.email)));
     const groupedResponse = [];
     for (const item of emails) {
+      const tmpPermissions = groupedPermissions[item];
       const data = {
         email: item,
-        role: groupedPermissions[item].length === matchRole ? ROLE.OWNER : ROLE.USER,
-        details: groupedPermissions[item]
+        name: tmpPermissions[0].name,
+        role: tmpPermissions.length === matchRole ? ROLE.OWNER : ROLE.USER,
```

Comment:
Can you use some other naming convention other than `tmpPermissions` this is confusing because, as per naming convention, I don't see any similarity between `tmpPermissions.length` and `matchRole`

Reply:
I think you want to verify if `existingAccess.length() === allAccess.length()`

Reply:
how about `matchRole` -> `role` and `tmpPermissions` -> `groupedPermissions` !?
@arkaprovob

Reply:
`tmpPermissions` does not sound right because it's not temporary permission; you are simply saving existing permissions into a variable and then comparing it with the total no of unique permissions in the table, dis I get that right?

Reply:
how about `allowedAccesses` instead of `Role` and `totalAccessess` instead of `tmpPermissions`  the idea is to give clarity to the developer who is reading your code.

Reply:
`role: allowedAccesses.length === totalAccessess.length ? ROLE.OWNER : ROLE.USER,`

Reply:
you can use `permission` instead of `access` if you want; So the final outcome would be
`role: allowedPermissions.length === allPermissions.length ? ROLE.OWNER : ROLE.USER,`

Reply:
Sure, I've changed it to allowedPermissions and allPermissions, please check.